packages/eslint-plugin/src/rules/class-name-casing.ts

```diff
@@ -78,19 +81,29 @@ module.exports = {
     //----------------------------------------------------------------------
 
     return {
-      'ClassDeclaration, TSInterfaceDeclaration, ClassExpression'(node) {
+      'ClassDeclaration, TSInterfaceDeclaration, ClassExpression'(
+        node:
+          | TSESTree.ClassDeclaration
+          | TSESTree.TSInterfaceDeclaration
+          | TSESTree.ClassExpression
+      ) {
         // class expressions (i.e. export default class {}) are OK
         if (node.id && !isPascalCase(node.id.name)) {
-          report(node);
+          report(node, node.id);
         }
       },
-      "VariableDeclarator[init.type='ClassExpression']"(node) {
-        const id = node.id;
+      "VariableDeclarator[init.type='ClassExpression']"(
+        node: TSESTree.VariableDeclarator
+      ) {
+        const id = node.id as TSESTree.Identifier;
```

Comment:
you should avoid doing typecasting
```
var { bar } = class {
	static bar() { return 2 }
}
```

Reply:
that code makes me sick!!! haha
I didn't even know that was valid code